src/main/java/org/waarp/openr66/database/data/DbConfiguration.java

```
@@ -180,32 +138,25 @@ protected String getUpdateAllFields() {
 
     @Override
     protected void setToArray() {
-        allFields[Columns.HOSTID.ordinal()].setValue(hostid);
-        allFields[Columns.READGLOBALLIMIT.ordinal()].setValue((readgloballimit / 10) * 10);
-        allFields[Columns.WRITEGLOBALLIMIT.ordinal()]
-                .setValue((writegloballimit / 10) * 10);
-        allFields[Columns.READSESSIONLIMIT.ordinal()]
-                .setValue((readsessionlimit / 10) * 10);
-        allFields[Columns.WRITESESSIONLIMIT.ordinal()]
-                .setValue((writesessionlimit / 10) * 10);
-        allFields[Columns.DELAYLIMIT.ordinal()].setValue(delayllimit);
-        allFields[Columns.UPDATEDINFO.ordinal()].setValue(updatedInfo);
+        allFields[Columns.HOSTID.ordinal()].setValue(limit.getHostid());
+        allFields[Columns.READGLOBALLIMIT.ordinal()].setValue(limit.getReadGlobalLimit());
+        allFields[Columns.WRITEGLOBALLIMIT.ordinal()].setValue(limit.getWriteGlobalLimit());
+        allFields[Columns.READSESSIONLIMIT.ordinal()].setValue(limit.getReadSessionLimit());
+        allFields[Columns.WRITESESSIONLIMIT.ordinal()].setValue(limit.getWriteSessionLimit());
+        allFields[Columns.DELAYLIMIT.ordinal()].setValue(limit.getDelayLimit());
+        allFields[Columns.UPDATEDINFO.ordinal()].setValue(limit.getUpdatedInfo());
     }
 
     @Override
     protected void setFromArray() throws WaarpDatabaseSqlException {
-        hostid = (String) allFields[Columns.HOSTID.ordinal()].getValue();
-        readgloballimit = (((Long) allFields[Columns.READGLOBALLIMIT.ordinal()]
-                .getValue()) / 10) * 10;
```

This is to get a rounf of 10 => Bug under Windows JVM where sleep or other elements using time with precision less than 10ms is not correct. So round to 10 (1024 => 1020) to ensure correctness.
However this is only really noticed on time being less than 20ms or so.

---

We could silently ignore this one...

---

I think the workaround is only useful on delay no ?

---

Yes